### ISSUE TITLE
MoneyFilterSearchModel currency method name wrong?

### ISSUE THREAD

**Issue by schleichardt** [sphereio/sphere-jvm-sdk#1046]
`io.sphere.sdk.search.model.MoneyFilterSearchModel#currency`

`io.sphere.sdk.products.search.ProductVariantFilterSearchModel#attribute`

@lauraluiz should't this be currencyCode?

**Comment by lauraluiz** [comment]
Depending on how you look at it. It is true that the backend name is `currencyCode`, but the method expects a currency (in particular an instance of `CurrencyUnit`). Currency code would only apply to a string like "EUR", while currency is a generic term that applies to any way to refer to a currency.

I see your point on wanting to mimic the API's naming, but my initial objective for the search DSL was hiding the complexity of the backend. Unfortunately after all transformations this is less and less true.

I need to review all methods to decide what makes more sense: backend or meaning naming.

**Comment by schleichardt** [comment]
also `io.sphere.sdk.products.search.ProductVariantFilterSearchModel#attribute` is not corresponding to the backend field name

**Comment by schleichardt** [comment]
this is annoying, because I did not find the method `attribute` because I did not use auto completion, instead I typed `attributes()`

**Comment by lauraluiz** [comment]
Actually we have:
- currencyCode -> currency
- centAmount -> before it was `amount` because it was expecting money, then I changed it to cent amount (long) for simplicity
- variants -> allVariants (to make it clear it is also including master variant)
- attributes -> attribute.ofString (otherwise attributes.ofString would be read wrong, and attributes alone would be missing the type)
- maybe more...

Trying to make it pure makes no sense for the developer. We created the DSL to make it easier to write, if we make it exactly as the API then it is easier for the developer to learn the search endpoint grammar and write it themselves. The SDL is meant for hiding the complexity of the API, not to share it.
